client/modules/IDE/reducers/project.js

```diff
@@ -12,40 +12,52 @@ const initialState = () => {
   };
 };
 
-const project = (state, action) => {
-  if (state === undefined) {
-    state = initialState(); // eslint-disable-line
+const projectSlice = createSlice({
+  name: 'project',
+  initialState,
+  reducers: {
+    setProjectName(state, action) {
+      state.name = action.payload;
+    },
+    newProject(state, action) {
+      const { id, name, updatedAt, owner } = action.payload;
+      state.id = id;
+      state.name = name;
+      state.updatedAt = updatedAt;
+      state.owner = owner;
+      state.isSaving = false;
+    },
+    setProject(state, action) {
+      const { id, name, updatedAt, owner } = action.payload;
+      state.id = id;
+      state.name = name;
+      state.updatedAt = updatedAt;
+      state.owner = owner;
+      state.isSaving = false;
+    },
+    resetProject(state) {
+      Object.assign(state, initialState);
```

Comment:
This isn't right because `initialState` is a function.  It can be written as `return initialState()`.